packages/lib-ts/src/intents/Swap.ts

```diff
@@ -405,7 +405,13 @@ export class Swap extends Intent {
     deadline: BigInt | null = null,
     nonce: string | null = null
   ) {
-    super(OperationType.Swap, settler, user, deadline, nonce)
+    let settlerAddress = settler
```

Comment:
Why not add all this logic in the Intent class instead of each instance?

Reply:
Because the Intent class does not receive chainId as a parameter

Reply:
Tbh it feels weird a findSettler static method in the Intent class

Reply:
Agree but I also don't like that you do it in every intent implementation, maybe we can pass the `settler` and the `chain id` to the intent and the Intent class is the one that does `settler ? settler : environment.getContext().findSettler(sourceChain)`